test/integration/extended-native-prototypes.js

```diff
@@ -0,0 +1,22 @@
+const mobx = require("../../src/mobx")
+
+describe("extended array prototype", () => {
+    const extensionKey = "__extension"
+
+    beforeAll(() => {
```

Comment:
It was intentional. I went with `All` variants because these set of tests are about mimicking a singular global environment change which is as dirty as it gets - the tests should reflect that truth with a single setup (and essentially a meaningless teardown)

Reply:
Alright, but a small comment would be appropriate in that case to get to the same thinking level :)

Reply:
No problem, I should of added that to begin with thanks!